lazy_synchornization_decision_making.py

```diff
@@ -0,0 +1,33 @@
+from node_response import NodeResponse
+import numpy as np
+
+class LazySynchornizationDecisionMaking:
+
+    def __init__(self, min_number_to_find_median=4):
+        self.min_number_to_find_median = min_number_to_find_median
+
+    def make_decision(self, nodes_responses: list):
+        # nodes_responses is a list of lists of node_response, where first list is a list od nodes,
+        # inner lists are nodes responses
+
+        dict = {}
+
+        for node_responses in nodes_responses:
+            for response in node_responses:
+                key = tuple([response.id, response.attribute_name, response.attribute_value])
+                if key not in dict:
```

Comment:
It is better to add an empty list if the key is not in the dict and append in all cases